src/backend/cuda/CMakeLists.txt

```diff
@@ -429,7 +433,8 @@ TARGET_LINK_LIBRARIES(afcuda    PRIVATE ${CUDA_CUBLAS_LIBRARIES}
                                 PRIVATE ${CUDA_cusparse_LIBRARY}
                                 PRIVATE ${CUDA_cusolver_LIBRARY}
                                 PRIVATE ${CUDA_nvvm_LIBRARY}
-                                PRIVATE ${CUDA_CUDA_LIBRARY})
+                                PRIVATE ${CUDA_CUDA_LIBRARY}
+                                PRIVATE ${Boost_LIBRARIES})
```

Comment:
I did a clean build and it worked, so i don't think boost is needed unless we would need shared_mutex in for any other purpose because that is not there in C++ until C++17 i think.